sql/thriftserver/pom.xml

```diff
@@ -0,0 +1,244 @@
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.spark</groupId>
+        <artifactId>spark-parent_2.12</artifactId>
+        <version>3.0.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>spark-thriftserver_2.12</artifactId>
+    <packaging>jar</packaging>
+    <name>Spark Project Thrift Server</name>
+    <url>http://spark.apache.org/</url>
+    <properties>
+        <sbt.project.name>thriftserver</sbt.project.name>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-hive_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-hive_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-cli</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-jdbc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${hive.group}</groupId>
+            <artifactId>hive-beeline</artifactId>
+        </dependency>
+        <dependency>
+        <groupId>${hive.group}</groupId>
+        <artifactId>hive-service</artifactId>
```

Comment:
Our `thriftserver` module is similar to `hive-service`. Could we adapt `hive-beeline` and `hive-jdbc` to this module?

Reply:
:+1: for doing that - to have our own client as a separate module, but not in the scope of this.

For now (in one of followup PR to this), we could start by pulling the client side into a new module (sql/jdbc-client?). I think `thriftserver` doesn't need to depend on `hive-service` and `hive-beeline`. These are needed only for a client.

Reply:
> Our `thriftserver` module is similar to `hive-service`. Could we adapt `hive-beeline` and `hive-jdbc` to this module?

Can't agree more to do this. Will make spark more independent, and we can have our own jdbc schema as `jdbc:spark` such as impala and presto.